addons/point_of_sale/static/src/app/models/related_models.js

```diff
@@ -474,652 +463,666 @@ export function createRelatedModels(modelDefs, modelClasses = {}, opts = {}) {
             };
         }
         return new Model({
-            models,
-            records,
-            model: models[model],
-            traps: { set: setTrapsCache[model] },
+            model,
+            traps: { set: setTrapsCache[modelName] },
         });
     }
 
-    const create = withoutProxyTrap(_create);
-    function _create(
-        crud,
-        model,
-        vals,
-        ignoreRelations = false,
-        fromSerialized = false,
-        delayedSetup = false
-    ) {
-        if (!("id" in vals)) {
-            vals["id"] = uuid(model);
+    /**
+     * A model (e.g. pos.order) points to an instance of this class.
+     */
+    class CRUD {
```

Comment:
An isEqual method can be very usefull here

Reply:
I put it in `Base` so we can just do `record.isEqual(otherRecord)`. No need to put here in CRUD.